tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/injected/ManagedThreadFactoryDefinitionOnEJBInjectedServlet.java

```diff
@@ -0,0 +1,247 @@
+    private UserTransaction tx;
+
+    @EJB
+    private ManagedThreadFactoryDefinitionInterface managedThreadFactoryDefinitionBean;
+
+    // Needed to initialize the ContextServiceDefinitions
+    @EJB
+    private ContextServiceDefinitionInterface contextServiceDefinitionBean;
+
+    @Resource(lookup = "java:app/concurrent/EJBThreadFactoryA")
+    private ManagedThreadFactory threadFactoryA;
+
+    @Resource(lookup = "java:comp/concurrent/EJBThreadFactoryB")
+    private ManagedThreadFactory threadFactoryB;
+
+    @Override
+    public void init() throws ServletException {
+        try {
+            managedThreadFactoryDefinitionBean.doLookup("java:module/concurrent/ContextB");
+        } catch (NamingException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    /**
+     * A ManagedThreadFactoryDefinition defined on an EJB with all attributes
+     * configured enforces priority and propagates context.
+     */
+    public void testManagedThreadFactoryDefinitionAllAttributesEJB() throws Throwable {
+        try {
+            IntContext.set(161);
+            StringContext.set("testManagedThreadFactoryDefinitionAllAttributesEJB-1");
+
+            IntContext.set(162);
+            StringContext.set("testManagedThreadFactoryDefinitionAllAttributesEJB-2");
+
+            Thread thread1 = threadFactoryA.newThread(() -> {
+            });
+            assertEquals(thread1.getPriority(), 4, "New threads must be created with the priority that is specified on "
+                    + "ManagedThreadFactoryDefinition");
+
+            BlockingQueue<Object> results = new LinkedBlockingQueue<>();
+
+            threadFactoryA.newThread(() -> {
+                results.add(Thread.currentThread().getPriority());
+                results.add(StringContext.get());
+                results.add(IntContext.get());
+                try {
+                    results.add(InitialContext.doLookup("java:app/concurrent/ContextA"));
+                } catch (Throwable x) {
+                    results.add(x);
+                }
+            }).start();
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(4),
+                    "ManagedThreadFactory must start threads with the configured priority.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), "",
+                    "Third-party context type StringContext must be cleared from thread "
+                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(161),
+                    "Third-party context type IntContext must be propagated to thread "
+                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration "
+                            + "based on the thread context at the time the ManagedThreadFactory was looked up.");
```

Comment:
IntContext is 0 (default) when looked up.
```suggestion
            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(0),
                    "Third-party context type IntContext must be propagated to thread "
                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration "
                            + "based on the thread context at the time the ManagedThreadFactory was looked up.");
```